### PR TITLE
refactor iop module skipping code

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
introduce
gboolean dt_iop_module_is_skipped(const struct dt_develop_t *dev, const struct dt_iop_module_t *module);

and make use of it for maintenance

### CODE REVIEW

File: src/develop/develop.c [hunk 1]
```diff
@@ -3187,10 +3185,8 @@ int dt_dev_distort_backtransform_locked(dt_develop_t *dev,
                && module->iop_order <= iop_order)
            || (transf_direction == DT_DEV_TRANSFORM_DIR_BACK_EXCL
                && module->iop_order < iop_order))
-       && !(dev->gui_module
-            && dev->gui_module != module
-            && (dev->gui_module->operation_tags_filter() & module->operation_tags())
-            && (pipe->type & (DT_DEV_PIXELPIPE_FULL | DT_DEV_PIXELPIPE_PREVIEW))))
+       && !(dt_iop_module_is_skipped(dev, module)
+            && (pipe->type & DT_DEV_PIXELPIPE_BASIC)))
```
Comment (TurboGit) [inline src/develop/develop.c:3189]: Isn't this a copy/paste mistake? Should be `pipe->type & (DT_DEV_PIXELPIPE_FULL | DT_DEV_PIXELPIPE_PREVIEW`?

File: src/develop/pixelpipe_cache.c [hunk 2]
```diff
@@ -136,11 +136,8 @@ static uint64_t _dev_pixelpipe_cache_basichash(
     dt_develop_t *dev = piece->module->dev;
 
     // don't take skipped modules into account
-    const gboolean skipped =
-      dev->gui_module
-      && dev->gui_module != piece->module
-      && (dev->gui_module->operation_tags_filter() & piece->module->operation_tags())
-      && (pipe->type & (DT_DEV_PIXELPIPE_FULL | DT_DEV_PIXELPIPE_PREVIEW));
+    const gboolean skipped = dt_iop_module_is_skipped(dev, piece->module)
+      && (pipe->type & DT_DEV_PIXELPIPE_BASIC);
```
Comment (TurboGit) [inline src/develop/pixelpipe_cache.c:140]: likewise

File: src/develop/develop.c (hunk 1 above)
Comment (jenshannoschwalm) [reply src/develop/develop.c:3189]: The binary or is already defined in pixel pipe.h...

File: src/develop/pixelpipe_cache.c (hunk 2 above)
Comment (jenshannoschwalm) [reply src/develop/pixelpipe_cache.c:140]: same as above. I started this pr while checking for consistency of "what modules are skipped ..."

File: src/develop/develop.c (hunk 1 above)
Comment (TurboGit) [reply src/develop/develop.c:3189]: I see, `BASIC` is `FULL | PREVIEW`, sorry for the noise!

**Pull request merged**